akka-actor-tests/src/test/scala/akka/io/dns/internal/AsyncDnsResolverSpec.scala

```diff
@@ -76,93 +71,77 @@ class AsyncDnsResolverSpec extends AkkaSpec(
         case Failure(ResolveFailedException(_)) ⇒
       }
     }
 
-    "gets SRV records if requested" in {
-      val dnsClient1 = TestProbe()
-      val dnsClient2 = TestProbe()
-      val r = resolver(List(dnsClient1.ref, dnsClient2.ref))
+    "gets SRV records if requested" in new Setup {
       r ! Resolve("cats.com", Srv)
       dnsClient1.expectMsg(SrvQuestion(1, "cats.com"))
       dnsClient1.reply(Answer(1, im.Seq.empty))
       dnsClient2.expectNoMessage()
-      expectMsg(Resolved("cats.com", im.Seq.empty))
+      senderProbe.expectMsg(Resolved("cats.com", im.Seq.empty))
     }
 
-    "response immediately IP address" in {
+    "response immediately IP address" in new Setup {
       val name = "127.0.0.1"
-      val dnsClient1 = TestProbe()
-      val r = resolver(List(dnsClient1.ref))
       r ! Resolve(name)
       dnsClient1.expectNoMessage(50.millis)
-      val answer = expectMsgType[Resolved]
+      val answer = senderProbe.expectMsgType[Resolved]
       answer.records.collect { case r: ARecord ⇒ r }.toSet shouldEqual Set(
         ARecord("127.0.0.1", Ttl.effectivelyForever, InetAddress.getByName("127.0.0.1"))
       )
     }
 
-    "response immediately for IPv6 address" in {
+    "response immediately for IPv6 address" in new Setup {
       val name = "1:2:3:0:0:0:0:0"
-      val dnsClient1 = TestProbe()
-      val r = resolver(List(dnsClient1.ref))
       r ! Resolve(name)
       dnsClient1.expectNoMessage(50.millis)
-      val answer = expectMsgType[Resolved]
+      val answer = senderProbe.expectMsgType[Resolved]
       val Seq(AAAARecord("1:2:3:0:0:0:0:0", Ttl.effectivelyForever, _)) = answer.records.collect { case r: AAAARecord ⇒ r }
     }
 
-    "return additional records for SRV requests" in {
-      val dnsClient1 = TestProbe()
-      val dnsClient2 = TestProbe()
-      val r = resolver(List(dnsClient1.ref, dnsClient2.ref))
+    "return additional records for SRV requests" in new Setup {
       r ! Resolve("cats.com", Srv)
       dnsClient1.expectMsg(SrvQuestion(1, "cats.com"))
       val srvRecs = im.Seq(SRVRecord("cats.com", Ttl.fromPositive(5000.seconds), 1, 1, 1, "a.cats.com"))
       val aRecs = im.Seq(ARecord("a.cats.com", Ttl.fromPositive(1.seconds), InetAddress.getByName("127.0.0.1")))
       dnsClient1.reply(Answer(1, srvRecs, aRecs))
       dnsClient2.expectNoMessage(50.millis)
-      expectMsg(Resolved("cats.com", srvRecs, aRecs))
+      senderProbe.expectMsg(Resolved("cats.com", srvRecs, aRecs))
 
       // cached the second time, don't have the probe reply
       r ! Resolve("cats.com", Srv)
-      expectMsg(Resolved("cats.com", srvRecs, aRecs))
+      senderProbe.expectMsg(Resolved("cats.com", srvRecs, aRecs))
     }
   }
 
   def resolver(clients: List[ActorRef]): ActorRef = {
     val settings = new DnsSettings(system.asInstanceOf[ExtendedActorSystem], ConfigFactory.parseString(
       """
           nameservers = ["one","two"]
-          resolve-timeout = 25ms
+          resolve-timeout = 150ms
```

Comment:
this is still rather tight on these CI machines, why does it have to be low?

Reply:
Just for test speed, we're testing that we fall back, I'll increase it a bit more